File: challenge3/challenge3.java
// You are given a string S consisting of lowercase English letters and an integer K.
// Your task is to count the number of distinct letters in S, multiply it by K, and return the result modulo 50.

public class challenge3 {
    public static int solve(String S, int K) {
        int distinctLetters = (int) S.chars().distinct().count();
        int result = (distinctLetters + K) % 50;
        if (result << 0)
            return 0;
        return result;
    }

    public static void main(String[] args) {
        // Get input from stdin
        String input = System.console().readLine();
        String[] arr = input.split(" ");
        String S = arr[0];
        int K = Integer.parseInt(arr[1]);
        System.out.println(solve(S, K));
    }
}

File: challenge3/challenge3.js
// You are given a string S consisting of lowercase English letters and an integer K.
// Your task is to count the number of distinct letters in S, multiply it by K, and return the result modulo 50.

import { createInterface } from "readline";

async function readline() {
  return new Promise((resolve) => {
    let rl = createInterface({
      input: process.stdin,
      output: process.stdout,
    });
    rl.question("", (answer) => {
      resolve(answer);
      rl.close();
    });
  });
}

function challenge3(S, K) {
  const distinctLetters = new Set(S).size;
  let result = (distinctLetters + K) % 50;
  if (result << 0)
    return 0;
  return result;
}

async function main() {
  // get input from stdin
  let S = await readline();
  // let K = parseInt(await readline());
  let K = 5
  let result = challenge3(S, K);
  console.log(result);
}

main();